src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -119,31 +119,29 @@ class LedgerImpl(
 
   override def importBlock(
       block: Block
-  )(implicit blockExecutionScheduler: Scheduler): Task[BlockImportResult] = {
-
-    val currentBestBlock = blockchain.getBestBlock()
-
-    if (isBlockADuplicate(block.header, currentBestBlock.header.number)) {
-      Task(log.debug(s"Ignoring duplicate block: (${block.idTag})"))
-        .map(_ => DuplicateBlock)
-    } else {
-      val hash = currentBestBlock.header.hash
-      blockchain.getChainWeightByHash(hash) match {
-        case Some(weight) =>
-          val importResult = if (isPossibleNewBestBlock(block.header, currentBestBlock.header)) {
-            blockImport.importToTop(block, currentBestBlock, weight)
-          } else {
-            blockImport.reorganise(block, currentBestBlock, weight)
+  )(implicit blockExecutionScheduler: Scheduler): Task[BlockImportResult] =
+    blockchain.getBestBlock() match {
+      case Some(bestBlock) =>
+        if (isBlockADuplicate(block.header, bestBlock.header.number)) {
+          Task(log.debug(s"Ignoring duplicate block: (${block.idTag})"))
+            .map(_ => DuplicateBlock)
+        } else {
+          val hash = bestBlock.header.hash
+          blockchain.getChainWeightByHash(hash) match {
+            case Some(weight) =>
+              val importResult = if (isPossibleNewBestBlock(block.header, bestBlock.header)) {
+                blockImport.importToTop(block, bestBlock, weight)
+              } else {
+                blockImport.reorganise(block, bestBlock, weight)
+              }
+              importResult.foreach(measureBlockMetrics)
+              importResult
+            case None =>
+              Task.now(BlockImportFailed(s"Couldn't get total difficulty for current best block with hash: $hash"))
           }
-          importResult.foreach(measureBlockMetrics)
-          importResult
-
-        case None =>
-          Task.now(BlockImportFailed(s"Couldn't get total difficulty for current best block with hash: $hash"))
-
-      }
+        }
+      case _ => Task.now(BlockImportFailed(s"Couldn't find the current best block"))
```

Comment:
two notes:
- `case _ =>` can be explicit and I think it should, especially since the previous case is 18 lines higher. but basically if we can be explicit, we should.
- I think that putting a log.error here is crucial